### ISSUE TITLE
Can't retrieve data from Firestore when `swcMinify: true`

### ISSUE THREAD

**Issue by kalwabed** [vercel/next.js#31391]
### What version of Next.js are you using?

12.0.3

### What version of Node.js are you using?

14

### What browser are you using?

Chrome

### What operating system are you using?

Linux

### How are you deploying your application?

Vercel

### Describe the Bug

I'm using Firebase version 9.4.1. I can't get the data by running a query to my Firebase Firestore. This only happens when in production and I enable `swcMinify` in the `next.config.js` file.

So, to **fix** it I just disable the `swcMinify` feature and everything is back to normal.

### Expected Behavior

Get data from my Firebase Firestore.

### To Reproduce

I took just one example that I got from the official Firebase(version 9) documentation to retrieve all data from collections: https://firebase.google.com/docs/firestore/query-data/get-data#get_all_documents_in_a_collection. Then I added a bit of how I implemented it:
```js
useEffect(() => {
	getUser(auth.id, user => {
		setUser(user)
	})
}, [auth])
```

In the `next.config.js` file we enable the `swcMinify` feature.
```js
module.exports = {
  swcMinify: true
  // ....
}
```

Then, you can try running it in production with `next build` then `next start`.

**Comment by Patrik-svobodik** [comment]
I'm facing the exact same issue, Firestore stops functioning when deployed to Vercel with swcMinify turned on

**Comment by timneutkens** [comment]
We've made some changes, could you try again using `next@canary`? `npm install next@canary`.

**Comment by kalwabed** [comment]
I want to try to implement the changes, but I'm facing another issue with Jotai after using `next@canary`.
![image](https://user-images.githubusercontent.com/49640654/147245525-a34c43a2-1d3d-4f86-80ad-b4355cb682cc.png)

**Comment by kalwabed** [comment]
I have tried and the issue is still there.

**Comment by balazsorban44** [comment]
Hi, we recently fixed some bugs in the minifier, and this `canary` has been updated.

#31753 had a clear reproduction so I could verify locally that it was fixed, and I, therefore, think this issue should also be good.

You can upgrade to at least `12.0.9-canary.4` or wait for it to be released to `latest`.

I'm going to close this, but please report back if you managed to try it out! :pray:

**Comment by kalwabed** [comment]
I confirm that this issue has been resolved in `12.0.9-canary.4`.
Thanks for all the hard work :pray:
